### PR TITLE
OD tf fixed get_gpu_name.

### PR DESCRIPTION
**Pull request by jakesabathia2**
Fix the crash for invalid conversion from Python side to vector<string>.

### CODE REVIEW

File: src/ml/neural_net/tf_compute_context.cpp [hunk 1]
```diff
@@ -261,17 +261,16 @@ tf_compute_context::tf_compute_context() = default;
...
   return 4294967296lu;
 }
 
 std::vector<std::string> tf_compute_context::gpu_names() const {
-  pybind11::object gpu_devices;
   std::vector<std::string> gpu_device_names;
 
   call_pybind_function([&]() {
-    
+    pybind11::object gpu_devices;
```
Comment (jakesabathia2) [inline src/ml/neural_net/tf_compute_context.cpp]: There is the change.

File: src/python/turicreate/toolkits/_tf_utils.py [hunk 2]
```diff
@@ -16,11 +16,11 @@ def get_gpu_names():
 	"""
 	import tensorflow as tf
 	gpu_names = tf.config.experimental.list_physical_devices('GPU')
-	return gpu_names
-	
+	return [str(gpu_name) for gpu_name in gpu_names]
```
Comment (jakesabathia2) [inline src/python/turicreate/toolkits/_tf_utils.py]: And there.

**Comment by jakesabathia2** [comment]
And my editor just auto correct all last blanks.
There are only two places which are really modified.

File: src/ml/neural_net/tf_compute_context.cpp (hunk 1 above)
Comment (shreyajain17) [reply src/ml/neural_net/tf_compute_context.cpp]: Can you add it like `pybind11::object gpu_devices = tf_gpu_devices.attr("get_gpu_names")();`

Comment (nickjong) [reply src/ml/neural_net/tf_compute_context.cpp]: In addition, let's fix the other instances in this file. Let's only declare `pybind11::object` variables inside the lambda being passed to `call_pybind_function`

File: src/ml/neural_net/tf_compute_context.cpp [hunk 3]
```diff
@@ -285,17 +284,18 @@ std::vector<std::string> tf_compute_context::gpu_names() const {
...
   call_pybind_function([&]() {
       pybind11::module tf_od_backend = pybind11::module::import(
           "turicreate.toolkits.object_detector._tf_model_architecture");
 
       // Make an instance of python object
-      object_detector = tf_od_backend.attr("ODTensorFlowModel")(h_in, w_in, n, c_out, weights, config);
+      pybind11::object object_detector = tf_od_backend.attr(
+          "ODTensorFlowModel")(h_in, w_in, n, c_out, weights, config);
+      result = std::unique_ptr<tf_model_backend>(
+          new tf_model_backend(object_detector));
```
Comment (nickjong) [inline src/ml/neural_net/tf_compute_context.cpp]: This can be simplified to `result.reset(new tf_model_backend(...))`

**Pull request merged**